src/activities/CategoriesEntity.js

```diff
@@ -122,18 +122,33 @@ export class CategoriesEntity extends Entity {
 		}
 	}
 
+	_generateDeletePreviousAndAssociateWithNewCategory(category, shouldDeleteDefaultCategory) {
+		const categoryEntity = this._getCategoryById(category.categoryId);
+		if (!categoryEntity) return;
+
+		const action = shouldDeleteDefaultCategory && categoryEntity.getActionByName(Actions.activities.categories.deletePreviousAndAssociateWithNewCategory);
+		if (!action) return;
+		const fields = [
+			{ name: 'deletePreviousCategory', value: shouldDeleteDefaultCategory }
+		];
+
+		return { action, fields };
+	}
+
 	_hasCategoryIdChanged(categoryId) {
 		const selectedCategory = this.getSelectedCategory();
 		const initialId = selectedCategory && selectedCategory.properties.categoryId;
 
 		return categoryId !== initialId;
 	}
 
-	async save(category) {
+	async save(category, shouldDeleteDefaultCategory = false) {
```

Comment:
Will update usages of the save to take the extra param, but setting it `false` here for safety.

Reply:
You mean we’re only updating the calls in discussion right? With the default parameter it shouldn’t affect existing usages